IdentityCore/src/oauth2/token/MSIDPrimaryRefreshToken.m

```diff
@@ -63,6 +57,14 @@ - (instancetype)initWithTokenCacheItem:(MSIDCredentialCacheItem *)tokenCacheItem
         _cachedAt = tokenCacheItem.cachedAt;
         _lastRecoveryAttempt = tokenCacheItem.lastRecoveryAttempt;
         _expiryInterval = [tokenCacheItem.expiryInterval integerValue];
+        
+        _externalKeyLocationType = (MSIDExternalPRTKeyLocationType)[[jsonDictionary msidObjectForKey:MSID_PRT_EXTERNAL_KEY_TYPE_CACHE_KEY ofClass:[NSString class]] integerValue];
+        
+        if (!_sessionKey && _externalKeyLocationType == MSIDExternalPRTKeyLocationTypeNone)
```

Comment:
previously we were returning early when session was nil, any reason why we moved it down ?

Reply:
Yes, because we want to store all attributes except sessionKey for PRTs that are associated with those other key types

Reply:
so for legacy keychain case do we still want to store all attributes ?

Reply:
Yes, but since it returns nil if sessionKey is nil anyway, then it doesn't really matter, no?